packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts

```diff
@@ -1655,34 +1593,26 @@ export class Cluster extends ClusterBase {
       throw new Error('Vpc must contain private subnets when public endpoint access is restricted');
     }
 
-    const placeClusterHandlerInVpc = props.placeClusterHandlerInVpc ?? false;
-
-    if (placeClusterHandlerInVpc && privateSubnets.length === 0) {
-      throw new Error('Cannot place cluster handler in the VPC since no private subnets could be selected');
-    }
-
-    if (props.clusterHandlerSecurityGroup && !placeClusterHandlerInVpc) {
-      throw new Error('Cannot specify clusterHandlerSecurityGroup without placeClusterHandlerInVpc set to true');
-    }
-
     if (props.serviceIpv4Cidr && props.ipFamily == IpFamily.IP_V6) {
       throw new Error('Cannot specify serviceIpv4Cidr with ipFamily equal to IpFamily.IP_V6');
     }
 
     this.authenticationMode = props.authenticationMode;
 
-    const resource = this._clusterResource = new ClusterResource(this, 'Resource', {
+    const resource = this._clusterResource = new CfnCluster(this, 'Resource', {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The Kubernetes API server for the Amazon EKS cluster is exposed if `endpointAccess` is set to `EndpointAccess.PUBLIC` in the `Cluster` resource or `endpointPublicAccess` is `true` in the `CfnCluster` resource. This configuration allows unauthorized internet access, increasing security risks such as data breaches and malicious activities. To secure the cluster, set `endpointAccess` to `EndpointAccess.PRIVATE` in the `Cluster` resource and `endpointPublicAccess` to `false` in the `CfnCluster` resource. Implement network access controls to restrict access to trusted sources within the AWS VPC. For more information refer to the documentation https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-eks.EndpointAccess.html.